cpp/src/gandiva/precompiled/arithmetic_ops.cc

```diff
@@ -157,7 +163,7 @@ boolean not_boolean(boolean in) { return !in; }
 NUMERIC_BOOL_DATE_FUNCTION(IS_DISTINCT_FROM)
 NUMERIC_BOOL_DATE_FUNCTION(IS_NOT_DISTINCT_FROM)
 
-#define DIVIDE(TYPE)                                               \
+#define DIVIDE_UNSAFE(TYPE)                                        \
```

Comment:
Seems like you swapped the meaning of SAFE/UNSAFE.

Reply:
In gandiva context, 

- functions that can returns errors are referred to "unsafe" - for such functions, gandiva has to check the validity bits before invoking the function. And, skip invoking the function if any of the arguments are invalid (to avoid spurious errors).
- functions that cannot returns errors are referred to "safe" - for such functions, gandiva can blindly invoke the function, without looking at the validity bits. These tend to perform better since there are no conditional checks.